app/lib/katello/resources/candlepin.rb

```diff
@@ -90,6 +90,18 @@ def self.name_to_key(a_name)
         def self.included_list(included)
           included.map { |value| "include=#{value}" }.join('&')
         end
+
+        def self.fetch_paged(page_size = 5000)
```

Comment:
Is 5000 the maximum page size allowed by CP? Is it the optimal page size?

Reply:
In my tests after 5000 the returns were marginal (it took about 208 secs ) while 7/8K ~ 195 secs.  
50K was ~ 3601 secs .The cp operations are very expensive and consume a lot of cpu/mem (pg = 91 % ,  9% java for 5k rows at a time).
I chose 5k as a happy medium.. But probably should have a setting for this